### ISSUE TITLE
Support for sp_image_*

### ISSUE THREAD

**Issue by todeveni** [Burgestrand/Hallon#10]
Support for image loading would be greatly appreciated.

If I only understood C, I would be more than willing to help implementing this.

**Comment by Burgestrand** [comment]
The main development branch is 1.0.0, and I’ve recently overcome one of the major issues I had with implementing these kind of things. It is very possible to make this in the 1.0.0 branch!

Images can only be loaded by providing an `image_id` (which is a series of bytes, a string if you wish). This `image_id` can be stored, but to get it initially you need either an `Album` or an `Artist` object, this means it’s not enough for me to just code the image implementation since you need another implementation just to use it!

So what needs to be in place is first the `Session` handling (working on this right now), and after that a way to retrieve either an `Album` or an `Artist`. Once this is done, you can fetch the image for the `Album` or `Artist`. So, how is this to be done, you ask? Well, the _absolutely_ quickest way to an image is implementing links, which can be used to retrieve an `Album` or `Artist` (or other objects such as playlists and tracks) directly!

However, depending on how you want to use this you might not want the quickest implementation. You can also convert a `Link` to a `Track` which will give you an `Album` or `Artist`. You can also convert a `Link` to a `Playlist`, which can give you tracks. Finally, you can retrieve a `Playlist` from your `PlaylistContainer` which is connected to your currently logged in user!

So, to give me a perspective, how would you want to use this? :)

PS: I can’t give you any promises of when it will be done. I’m a student and I work on Hallon only when I have time and motivation. What I can promise you is that I’ll work on image support earlier than I had intended before you posted this issue ;)

**Comment by Burgestrand** [comment]
Spotify added support for directly linking to images with Spotify URIs in libspotify `v0.0.8` that was released about ~3 weeks ago, and this functionality is now added to Hallon.

As for the test, I actually retrieved the image URI using the raw API. Support for other ways of retrieving the image is in the queue.

Callbacks aren’t supported yet (don’t need them), but I’ll create a separate issue for them.
